Documentation/ApiOverview/LinkBrowser/Linkhandler/CustomLinkHandlers.rst

```diff
@@ -0,0 +1,202 @@
+            ->loadRequireJsModule('TYPO3/CMS/Examples/GitHubLinkHandler');
+
+         $this->view->assign('project', $this->configuration['project']);
+         $this->view->assign('action', $this->configuration['action']);
+         $this->view->assign('github', !empty($this->linkParts) ? $this->linkParts['url']['github'] : '');
+         return $this->view->render('GitHub');
+      }
+
+      /**
+      * @return string[] Array of body-tag attributes
+      */
+      public function getBodyTagAttributes(): array
+      {
+         return [];
+      }
+
+      /**
+      * @return array
+      */
+      public function getLinkAttributes()
+      {
+         return $this->linkAttributes;
+      }
+
+      /**
+      * @param string[] $fieldDefinitions Array of link attribute field definitions
+      * @return string[]
+      */
+      public function modifyLinkAttributes(array $fieldDefinitions)
+      {
+         return $fieldDefinitions;
+      }
+
+      /**
+      * We don't support updates since there is no difference to simply set the link again.
+      *
+      * @return bool
+      */
+      public function isUpdateSupported()
+      {
+         return FALSE;
+      }
+   }
+
+The LinkHandler then has to be registered via page TSCONFIG:
+
+.. code-block:: typoscript
+
+   TCEMAIN.linkHandler {
+      github {
+         handler = T3docs\\Examples\\LinkHandler\\GitHubLinkHandler
+         label = LLL:EXT:examples/Resources/Private/Language/locallang_browse_links.xlf:github
+         displayAfter = url
+         scanBefore = url
+         configuration {
+             project = TYPO3-Documentation/TYPO3CMS-Reference-CoreApi
+             action = issues
+         }
+      }
+   }
+
+
+And the JavaScript, depending on :ref:`requirejs`, has to be added in a file
+:file:`examples/Resources/Public/JavaScript/GitHubLinkHandler.js`:
+
+.. code-block:: javascript
+
+   /**
+    * Module: TYPO3/CMS/Examples/GitHubLinkHandler
+    * Github issue link interaction
+    */
+   define(['jquery', 'TYPO3/CMS/Recordlist/LinkBrowser'], function($, LinkBrowser) {
+      'use strict';
+
+      /**
+       *
+       * @type {{}}
+       * @exports T3docs/Examples/GitHubLinkHandler
+       */
+      var GitHubLinkHandler = {};
+
+      $(function() {
+         $('#lgithubform').on('submit', function(event) {
+            event.preventDefault();
+
+            var value = $(this).find('[name="lgithub"]').val();
+            if (value === 'github:') {
+               return;
+            }
+            if (value.indexOf('github:') === 0) {
+               value = value.substr(7);
+            }
+            LinkBrowser.finalizeFunction('github:' + value);
+         });
+      });
+
+      return GitHubLinkHandler;
+   });
+
+This would create a link looking like this:
+
+.. code-block:: html
+
+   <a href="github:123">Example Link</a>
+
+Which could for example be interpreted by a custom protocol handler on a
+company computers operating system.
```

Comment:
```suggestion
Which could, for example, be interpreted by a custom protocol handler on a
company computer's operating system.
```

Reply:
Also wondering if it might be helpful to mention the use case further up (to give context to the example), rather than at the bottom?